syntax = "proto3";

package hi_hello;

service GreetingService {
    rpc SayHi (HiRequest) returns (HiResponse) {}
    rpc SayHello (HelloRequest) returns (HelloResponse) {}
    rpc FindUsbDevice (FindUsbDeviceRequest) returns (FindUsbDeviceResponse) {}
    rpc SetDeviceSpeed (SpeedRequest) returns (SpeedResponse) {}
    rpc SendData (DataRequest) returns (DataResponse) {}
    rpc CdStressOn (CdStressRequest) returns (CdStressResponse) {} // Existing RPC definition
    rpc CdStressOff (CdStressOffRequest) returns (CdStressOffResponse) {} // New RPC definition
    rpc SendPRswapCommand (PRswapRequest) returns (PRswapResponse) {} // PRswap RPC definition
    rpc SendDRswapCommand (DRswapRequest) returns (DRswapResponse) {} // DRswap RPC definition
    rpc PdCaptiveCable (PdCaptiveCableRequest) returns (PdCaptiveCableResponse) {}
    rpc PdChargerPort (PdChargerPortRequest) returns (PdChargerPortResponse) {}
}

message HiRequest {
    string name = 1;
}

message HiResponse {
    string message = 1;
}

message HelloRequest {
    string name = 1;
}

message HelloResponse {
    string message = 1;
}

message FindUsbDeviceRequest {
    bool list_devices = 1;
}

message UsbDevice {
    string manufacturer = 1;
    string product = 2;
    uint32 firmware_version = 3;
}

message FindUsbDeviceResponse {
    repeated UsbDevice devices = 1;
}

message SpeedRequest {
    string speed_type = 1;
}

message SpeedResponse {
    string message = 1;
}

message DataRequest {
    int32 watts = 1;
}

message DataResponse {
    string message = 1;
}

message CdStressRequest {
    // Define any fields needed for CdStressOn
}

message CdStressResponse {
    string message = 1;
}

message CdStressOffRequest {
    // Define any fields needed for CdStressOff
}

message CdStressOffResponse {
    string message = 1;
}

message PRswapRequest {
    bytes data = 1;
}

message PRswapResponse {
    string message = 1;
}

message DRswapRequest {
    bytes data = 1;
}

message DRswapResponse {
    string message = 1;
}

message PdCaptiveCableRequest {
    // Define any fields needed for PdCaptiveCable
}

message PdCaptiveCableResponse {
    string message = 1;
}

message PdChargerPortRequest {
    // Define any fields needed for PdChargerPort
}

message PdChargerPortResponse {
    string message = 1;
}